pages/products/index.tsx

```diff
@@ -30,21 +30,19 @@ type ProductListingProps = Readonly<{
 }>
 
 const createQueryStringUrl = (params: Record<string, string | string[] | undefined>) =>
-  Object.keys(params).length > 0 ?
-    Object.entries(params).reduce(
-      (prev, next, index) => {
-        const [paramKey, paramValue] = next;
+  Object.keys(params).length > 0 ? '?'+
+    Object.entries(params).map(
```

Comment:
Although I generally like avoiding a function body when it is not necessary, in this case, I would be in favour of putting the `Object.entries()...` expression into a local constant for better readability. It is not so easy to work out the different parts of the expression in my opinion.